account_invoice_report_grouped_by_picking/models/account_move.py

```diff
@@ -15,7 +15,12 @@ class AccountMove(models.Model):
     @api.model
     def _sort_grouped_lines(self, lines_dic):
         return sorted(
-            lines_dic, key=lambda x: (x["picking"].date, x["picking"].date_done)
+            lines_dic,
+            key=lambda x: (
+                x["picking"].date_done
+                and (x["picking"].date, x["picking"].date_done)
+                or (x["picking"].date, x["picking"].date)
```

Comment:
Another option:
```suggestion
                or x["picking"].date
```

Reply:
This one wouldn't be valid as we would have a mix of tuples and datetimes.